build.sbt

```diff
@@ -2348,10 +2348,18 @@ lazy val `engine-runner` = project
             "-Dnic=nic"
           ),
           mainClass = Some("org.enso.runner.Main"),
-          additionalCp = Seq(
-            "runtime.jar",
-            "runner.jar"
-          ),
+          additionalCp = () => {
+            val core = Seq(
+              "runtime.jar",
+              "runner.jar"
+            )
+            val jars = `base-polyglot-root`.listFiles("*.jar")
```

Comment:
That's really just delaying the problem but not solving it. It should be a proper dependency. I will take a look later today.

Reply:
OK. However there is a proper [dependsOn](https://github.com/enso-org/enso/pull/9866/files#diff-5634c415cd8c8504fdb973a3ed092300b43c4b8fc1e184f7249eb29a55511f91R2382). E.g. at the end the system will behave correctly.

Reply:
You are probably right about the `dependsOn`. But still, I think there should be an assertion that all these JAR archives exists, and that the `base-polyglot-root` directory is not empty. A simple `assert` should be fine, in case of assertion error, sbt will display it.